tests/e2e/monitoring/test_monitoring_on_negative_scenarios.py

```diff
@@ -0,0 +1,426 @@
+            )
+
+            # Validate prometheus pod is re-spinned on new healthy node
+            pod_info = pod_obj.get()
+            new_node = pod_info['spec']['nodeName']
+            assert new_node not in prometheus_node, (
+                'Promethues pod not re-spinned on new node'
+            )
+            log.info(f"Prometheus pod re-spinned on new node {new_node}")
+
+            # Validate same pvc is mounted on prometheus pod
+            assert pod_info['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] in pvc_name, (
+                f"Old pvc not found after restarting the prometheus pod {pod_obj.name}"
+            )
+
+            # Validate the prometheus health is ok
+            assert prometheus_health_check(), (
+                "Prometheus cluster health is not OK"
+            )
+
+            # Mark the nodes back to schedulable
+            schedule_nodes([prometheus_node])
+
+        # Check the node are Ready state and check cluster is health ok
+        self.sanity_helpers.health_check()
+
+        # Check for the created pvc metrics after rebooting the master nodes
+        for pod_obj in pods:
+            assert check_pvcdata_collected_on_prometheus(pod_obj.pvc.name), (
+                f"On prometheus pod for created pvc {pod_obj.pvc.name} related data is not collected"
+            )
+
+    @pytest.mark.polarion_id("OCS-580")
+    def test_monitoring_after_respinning_ceph_pods(self, pods):
+        """
+        Test case to validate respinning the ceph pods and
+        its interaction with prometheus pod
+
+        """
+
+        # Re-spin the ceph pods(i.e mgr, mon, osd, mds) one by one
+        resource_to_delete = ['mgr', 'mon', 'osd']
+        disruption = Disruptions()
+        for res_to_del in resource_to_delete:
+            disruption.set_resource(resource=res_to_del)
+            disruption.delete_resource()
+
+        # Check for the created pvc metrics on prometheus pod
+        for pod_obj in pods:
+            assert check_pvcdata_collected_on_prometheus(pod_obj.pvc.name), (
+                f"On prometheus pod for created pvc {pod_obj.pvc.name} related data is not collected"
+            )
+
+    @pytest.mark.polarion_id("OCS-605")
+    def test_monitoring_when_osd_down(self, pods):
+        """
+        Test case to validate monitoring when osd is down
+
+        """
+
+        # Get osd pods
+        osd_pod_list = pod.get_osd_pods()
+
+        # Make one of the osd down(first one)
+        ocp_obj = ocp.OCP(kind=constants.DEPLOYMENT, namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+
+        params = '{"spec": {"replicas": 0}}'
+        resource_name = osd_pod_list[0].get().get('metadata').get('name')
+        resource_name = '-'.join(resource_name.split('-')[0:4])
+        assert ocp_obj.patch(resource_name=resource_name, params=params), (
+            f"Failed to change the replica count of osd {resource_name} to 0"
+        )
+
+        # Validate osd is down
```

Comment:
pls consider making osd pod down and up actions into a separate function (maybe as a separate PR).